### PR TITLE
docs(theming): add Under the Hood section

### PR DESCRIPTION
**Pull request by sscovil**
document the inner workings of theme generation, per discussion in #980

### CODE REVIEW

File: docs/content/Theming/03_configuring_a_theme.md [hunk 1]
```diff
@@ -132,3 +132,28 @@ angular.module('myApp', ['ngMaterial'])
...
 </hljs>
+
+### Under the Hood
+
+Angular Material dynamically generates CSS for registered themes by injecting several
+`<script>` tags into the `<head>` section of the application at runtime. Here is how
+that process currently works:
+
+1. During the Angular Material build process, a function in `gulp/util.js` called
+[themeBuildStream](https://github.com/angular/material/blob/master/gulp/util.js#L187)
```
Comment (ThomasBurleson) [inline docs/content/Theming/03_configuring_a_theme.md]: I do not believe `default-theme.scss` exists anymore.

File: docs/content/Theming/03_configuring_a_theme.md [hunk 2]
```diff
@@ -132,3 +132,28 @@ angular.module('myApp', ['ngMaterial'])
...
+compiles all `*-theme.scss` files from components into a single CSS string that is
+injected into the `material.core` module as a constant called `$MD_THEME_CSS`.
+
+1. At runtime, a function in `src/core/services/theming/theming.js` called
```
Comment (ThomasBurleson) [inline docs/content/Theming/03_configuring_a_theme.md]: I need to confirm... but I believe we changed our rules so **default** components styles do not have `.md-default-theme` selectors in the CSS. This allows the components to be more easily styled. 

@rschmukler - can you confirm?

File: docs/content/Theming/03_configuring_a_theme.md (hunk 1 above)
Comment (sscovil) [inline docs/content/Theming/03_configuring_a_theme.md]: The files I'm referring to are in each component directory (e.g. `autocomplete-theme.scss`, `backdrop-theme.scss`, `bottom-sheet-theme.scss`).

File: docs/content/Theming/03_configuring_a_theme.md (hunk 2 above)
Comment (sscovil) [inline docs/content/Theming/03_configuring_a_theme.md]: Not sure if that is the case, but the default theme definitely produces 16 style tags in the head section. Here is a plunk with a build of the latest master commit: http://plnkr.co/edit/5W0q58xmvg76TSWmf32W?p=preview

Comment (ThomasBurleson) [inline docs/content/Theming/03_configuring_a_theme.md]: @rschmukler - did we not make some perf changes to Theme injection regarding the default themes? Or was that only to not namespacing the default CSS styles ?

**Pull request closed without merging**